includes/class-wc-cart-totals.php

```diff
@@ -226,7 +226,7 @@ protected function get_items_from_cart() {
 			$item->taxable                 = 'taxable' === $cart_item['data']->get_tax_status();
 			$item->price_includes_tax      = wc_prices_include_tax();
 			$item->quantity                = $cart_item['quantity'];
-			$item->price                   = wc_add_number_precision_deep( $cart_item['data']->get_price() * $cart_item['quantity'] );
+			$item->price                   = wc_add_number_precision_deep( (float) $cart_item['data']->get_price() * (float) $cart_item['quantity'] );
```

Comment:
:memo: Choice of a cast to float for the quantity (instead of a cast to integer) is deliberate. The test suite contains cases where we use quantities such as 0.5.